mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -229,7 +229,8 @@ public void save(Mailbox mailbox) throws MailboxException {
             }
             folder.setACL(session, mailbox.getACL());
         }
-        
+        cacheMailbox(mailbox);
```

Comment:
cacheMailbox is the operation that add the mailboxId.

It was called on the first message operation on the mailbox.